reaktive/src/jsMain/kotlin/com/badoo/reaktive/timer/Timers.kt

```diff
@@ -0,0 +1,21 @@
+package com.badoo.reaktive.timer
+
+import kotlin.browser.window
+import com.badoo.reaktive.timer.external.clearInterval as clearIntervalExternal
+import com.badoo.reaktive.timer.external.clearTimeout as clearTimeoutExternal
+import com.badoo.reaktive.timer.external.setInterval as setIntervalExternal
+import com.badoo.reaktive.timer.external.setTimeout as setTimeoutExternal
+
+private val isWindowDefined: Boolean = jsTypeOf(window) != "undefined"
```

Comment:
This is verbose, could be better. However I tried to check if `setTimeout` is undefined but could not.

@lukaville Any idea how could it be done?

Reply:
I checked the docs and looks like the standard way to access global object in all environments is `globalThis`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/globalThis
https://www.codementor.io/@robertwozniak/to-clean-it-up-globalthis-is-here-x252w3rhh

It works in browsers and returns window:
```
> globalThis
Window {parent: Window, opener: null, top: Window, length: 0, frames: Window, …}
```

And it works in NodeJS as well:
```
> globalThis
Object [global] {
  global: [Circular],
  clearInterval: [Function: clearInterval],
  clearTimeout: [Function: clearTimeout],
  setInterval: [Function: setInterval],
  setTimeout: [Function: setTimeout] { [Symbol(util.promisify.custom)]: [Function] },
  queueMicrotask: [Function: queueMicrotask],
  clearImmediate: [Function: clearImmediate],
  setImmediate: [Function: setImmediate] {
    [Symbol(util.promisify.custom)]: [Function]
  }
}
```

Reply:
Awesome, let me try that and get back to you

Reply:
```
external val globalThis: dynamic

internal fun setTimeout(task: () -> Unit, delayMillis: Int): Int =
    globalThis.setTimeout(task, delayMillis) as Int
```

using `globalThis.setTimeout` should return `Int` but if I try this it throws ClassCastException on nodejs
<img width="595" alt="Screen Shot 2020-06-18 at 3 10 06 PM" src="https://user-images.githubusercontent.com/20396931/85016280-7ed16500-b17a-11ea-80ee-283529e29e13.png">

Did I define it wrong here?
@lukaville

Reply:
@lukaville Fixed it in the latest commit, resolving. Thanks for the suggestion